snapcraft/internal/meta/snap.py

```diff
@@ -317,6 +319,31 @@ def from_dict(cls, snap_dict: Dict[str, Any]) -> "Snap":
 
         layout = snap_dict.pop("layout", None)
         license = snap_dict.pop("license", None)
+
+        # snap.yaml will have links, snapcraft.yaml the top level entries.
+        # some considerations:
+        # - links should not be allowed in snapcraft.yaml by the schema.
+        # - the top level entries would never exist in snap.yaml
+        #   (snap pack would reject them).
```

Comment:
we have no code to do that in my current PR, we tend not to refuse extra fields but maybe in this case we should

Reply:
thanks, this is from the context of a snap.yaml generated by snapcraft. I will remove the "snap pack" will reject them comment as you are right, that slipped my mind